packages/app/universal-ts-utils/src/object/areEquals.ts

```diff
@@ -0,0 +1,46 @@
+/**
+ * Determines if two unknown values are deeply equal. This function handles primitive types,
+ * arrays, and objects. For arrays and objects, it performs a recursive equality check.
+ *
+ * @param {unknown} a - The first value to compare. Can be of any type including array and object.
+ * @param {unknown} b - The second value to compare. Should be of the same type or comparable with `a`.
+ * @returns {boolean} Returns `true` if the two values are deeply equal, `false` otherwise.
+ *
+ * @example
+ * ```typescript
+ * areEquals(1, 1) // true
+ * areEquals([1, 2], [1, 2]) // true
+ * areEquals({ name: 'John' }, { name: 'John' }) // true
+ * areEquals(null, null) // true
+ * areEquals(undefined, null) // false
+ * areEquals([1, [2, 3]], [1, [2, 3]]) // true
+ * areEquals([{ id: 1 }], [{ id: 1 }]) // true
+ * ```
+ */
+export const areEquals = (a: unknown, b: unknown): boolean => {
```

Comment:
🟢 Just my opinion.

This could be split into three separate functions - isDeepEqual, isShallowEqual and isStrictEqual and then compose those as needed (base on use case). Currently, this function does a bit of everything.

Reply:
So, what should we do with this? 😓🙏

Reply:
If it is doing deep equal, then I'm in favor of being more explicit and naming the function "isDeepEqual". It makes it much more obvious for the reader. And knowing whether the function is doing shallow/deep equal is essential.

Reply:
Makes total sense to me, I am going to rename it and maybe we can implement the `isStrictEqual` in the future if we see it is needed 🙏